ios/COVIDSafePaths/AppDelegate.m

```diff
@@ -78,15 +79,25 @@ - (void)application:(UIApplication *)application didReceiveLocalNotification:(UI
 }
 
 - (void)applicationWillTerminate:(UIApplication *)application {
-  UILocalNotification *notification = [[UILocalNotification alloc] init];
-  notification.fireDate = [NSDate dateWithTimeIntervalSinceNow:0];
-  notification.alertTitle = NSLocalizedString(@"ios.app_closed_alert_title", @"Title of notification when app is closed");
-  notification.alertBody = NSLocalizedString(@"ios.app_closed_alert_text", @"Body text of notification when app is closed");
-  notification.timeZone = [NSTimeZone defaultTimeZone];
-  notification.soundName = UILocalNotificationDefaultSoundName;
-  notification.applicationIconBadgeNumber = 0;
-
-  [RNCPushNotificationIOS didReceiveLocalNotification:notification];
+  //Show local notifiation at first time only.
+  if(![[NSUserDefaults standardUserDefaults] boolForKey:@"NotificationAtFirstTimeOnly"]) {
+    if ([[UIApplication sharedApplication] respondsToSelector:@selector(currentUserNotificationSettings)]){
+      //Checking local notification permission or not.
+      UIUserNotificationSettings *grantedSettings = [[UIApplication sharedApplication] currentUserNotificationSettings];
+      if (grantedSettings.types != UIUserNotificationTypeNone){
+        // Set local notification.
+        UILocalNotification *_localNotification = [[UILocalNotification alloc] init];
```

Comment:
Great, looking good. 

This whole method is getting large and there are mixed levels of abstraction. You've added good comments, yet I think those would be better as separate methods:

(pseudo code)

```js
applicationWillTerminate() {
  if (isFirstTimeClosing() && hasNotificationPermissions()) {
    notifThatWeAreStillTracking();  // <--- if this is moved to a new method the applicationWillTerminate method is easier to understand
  }
}
```

Reply:
I will do.